graph/oval_graph.py

```diff
@@ -161,75 +161,52 @@ def change_tree_value(self, node_id, value):
+        if self.node_type == 'value':
+            return \
+                str(self.node_id).\
+                replace('xccdf_org.ssgproject.content_rule_', '').\
+                replace('oval:ssg-', '').\
+                replace(':def:1', '').\
+                replace(':tst:1', '').\
+                replace('test_', '')
+        else:
+            if str(self.node_id).startswith(
+                    'xccdf_org.ssgproject.content_rule_'):
+                return \
+                    str(self.node_id).\
+                    replace('xccdf_org.ssgproject.content_rule_', '')
+            return self.value
+
+    def _get_color_node(self):
         if self.value == 'true':
-            return {
-                'id': self.node_id,
-                'label': self._get_label(),
-                'url': 'null',
-                'text': 'null',
-                'title': self.node_id,
-                "x": x,
-                "y": y,
-                "size": 3,
-                "color": '#00ff00'}
+            return '#00ff00'
         elif self.value == 'false':
-            return {
-                'id': self.node_id,
-                'label': self._get_label(),
-                'url': 'null',
-                'text': 'null',
-                'title': self.node_id,
-                "x": x,
-                "y": y,
-                "size": 3,
-                "color": '#ff0000'}
+            return '#ff0000'
         else:
             if self.evaluate_tree() == 'true':
-                return {
-                    'id': self.node_id,
-                    'label': self.value,
-                    'url': 'null',
-                    'text': 'null',
-                    'title': self.node_id,
-                    "x": x,
-                    "y": y,
-                    "size": 3,
-                    "color": '#00ff00'
-                }
+                return '#00ff00'
             elif self.evaluate_tree() == 'false':
-                return {
-                    'id': self.node_id,
-                    'label': self.value,
-                    'url': 'null',
-                    'text': 'null',
-                    'title': self.node_id,
-                    "x": x,
-                    "y": y,
-                    "size": 3,
-                    "color": '#ff0000'
-                }
+                return '#ff0000'
             else:
-                return {
-                    'id': self.node_id,
-                    'label': self.value,
-                    'url': 'null',
-                    'text': 'null',
-                    'title': str(self.node_id) + ' ' + self.value,
-                    "x": x,
-                    "y": y,
-                    "size": 3,
-                    "color": '#000000'
-                }
+                return '#000000'
+
+    def _get_node_title(self):
+        if self._get_color_node() == '#000000':
+            return str(self.node_id) + ' ' + self.value
```

Comment:
I don't like this condition. The condition means: If the node is black then add its value. However, the black color is only a consequence of a different property.

Even more, it is hard to know what is the meaning of the black color. I can't figure out when this condition is true without running the code and seeing the output. Is it the self.value is non-empty? Or when does it happen?

What if in future you decide to use a different color? Then you would have to remember that you have to change this code as well. I'm sure you will forget about it and then you will wonder why it doesn't work and you might spend a lot of time investigating it.

I believe you should make the behavior obvious for anyone who reads the code. Please think how to rewrite the condition so that it doesn't depend on a color but it directly checks some property of the class.